src/app/helpers/jwt.interceptor.spec.ts

```diff
@@ -1,45 +1,76 @@
-import {TestBed} from '@angular/core/testing';
-import {HttpClientTestingModule} from '@angular/common/http/testing';
-import {httpHandlerSpy, httpRequestSpy} from './spies';
-import {HttpHandler, HttpRequest} from '@angular/common/http';
-import {of} from 'rxjs';
-import {JwtInterceptor} from './jwt.interceptor';
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { httpHandlerSpy, httpRequestSpy } from './spies';
+import { HttpHandler, HttpRequest } from '@angular/common/http';
+import { of } from 'rxjs';
+import { JwtInterceptor } from './jwt.interceptor';
+import { AuthService } from '../services/auth.service';
 
 
 describe('JWTInterceptor', () => {
   let jwtInterceptor: JwtInterceptor;
+  let authService: AuthService;
+  let mockAuthService;
+  let service;
+  let spy;
+  let req;
+  let authServiceData;
+
+  authServiceData = {
+    currentUser: {
+      token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpZCI6MTAsImVtYWlsIjoib2dyZUB2aW1haWwyNC5jb20iLCJleHAiOjE1NjY2MzQxMDB9.5u2ylra-VwFb5qDjVsTVfcXPlGZlr9bGg54YyCwvkDA'
+    },
+    isLoggedIn: () => true
+  }
 
   beforeEach(() => {
-    jwtInterceptor = new JwtInterceptor();
+    jwtInterceptor = new JwtInterceptor(authService);
 
     TestBed.configureTestingModule({
       providers: [
         JwtInterceptor,
-        {provide: HttpRequest, useValue: httpRequestSpy},
-        {provide: HttpHandler, useValue: httpHandlerSpy},
+        { provide: HttpRequest, useValue: httpRequestSpy },
+        { provide: HttpHandler, useValue: httpHandlerSpy },
       ],
       imports: [HttpClientTestingModule]
     });
   });
 
+  const setUp = (authService: AuthService) => {
+    spy = jasmine.createSpyObj('HttpHandler', ['handle']);
```

Comment:
For uniformity, we can have this jasmine spy in the `src/app/helpers/spies.ts` file

Reply:
I like this, this also was resolved. Thanks @Tevinthuku